imap-proto/src/parser/rfc3501.rs

```diff
@@ -166,16 +166,48 @@ named!(resp_text_code<ResponseCode>, do_parse!(
     (coded)
 ));
 
-named!(capability<&str>, do_parse!(
-    tag_s!(" ") >>
-    atom: map_res!(take_till1_s!(atom_specials), str::from_utf8) >>
-    (atom)
+/// [RFC3501 - 9. Formal Syntax - capability](https://tools.ietf.org/html/rfc3501#section-9)
+named!(capability_auth<Capability>, do_parse!(
+    tag_s!("AUTH=") >>
+    auth: atom >>
+    (Capability::Auth(auth))
+));
+
+/// [RFC3501 - 9. Formal Syntax - capability](https://tools.ietf.org/html/rfc3501#section-9)
+named!(capability_atom<Capability>, do_parse!(
+    not!(tag!("IMAP4rev1")) >>
+    atom: atom >>
+    (Capability::Atom(atom))
+));
+
+/// [RFC3501 - 9. Formal Syntax - capability](https://tools.ietf.org/html/rfc3501#section-9)
+named!(capability<Capability>, do_parse!(
+    tag!(" ") >>
+    capability: alt!(
+        capability_auth |
+        capability_atom
+    ) >>
+    (capability)
 ));
 
+/// The CAPABILITY command requests a listing of capabilities that the
+/// server supports.
+/// [RFC3501 - 6.1.1 CAPABILITY Command](https://tools.ietf.org/html/rfc3501#section-6.1.1)
+/// [RFC3501 - 9. Formal Syntax - capability-data](https://tools.ietf.org/html/rfc3501#section-9)
```

Comment:
I'm not a fan of repeating the URL of the formal syntax section in each comment here -- it's very wordy and has very limited added value. (Same for other parsers.)